nomad/structs/workload_id.go

```diff
@@ -40,3 +85,80 @@ func (wi *WorkloadIdentity) Equal(other *WorkloadIdentity) bool {
 
 	return true
 }
+
+func (wi *WorkloadIdentity) Canonicalize() {
+	if wi == nil {
+		return
+	}
+
+	if wi.Name == "" {
+		wi.Name = WorkloadIdentityDefaultName
+	}
+
+	// The default identity is only valid for use with Nomad itself.
+	if wi.Name == WorkloadIdentityDefaultName {
+		wi.Audience = []string{WorkloadIdentityDefaultAud}
+	}
```

Comment:
AFAICT audiences are pure anarchy of string matching. [GKE seems to use a full URL](https://cloud.google.com/blog/products/containers-kubernetes/kubernetes-bound-service-account-tokens) which seems wise, and [Vault implies a full URL is a reasonable audience in at least one spot](https://developer.hashicorp.com/vault/docs/auth/jwt). Other examples often use basic service names like [`"vault"`](https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/), and [Consul just throws some random ascii in there.](https://developer.hashicorp.com/consul/docs/security/acl/auth-methods/jwt)

So yeah, you could setup a 3rd party service that accepts JWT with `aud="nomadproject.io"`, but you could also just not validate the audience at all.

`</rant>`

The short answer is that: yes, that would be an incorrect configuration of those third-parties. It would allow that third party to masquerade as your service to the real Nomad API, and do it for the lifetime of the allocation since we won't be able to enforce expirations on default identities for a couple releases!

If your mind has started wondering `what if a user has multiple nomad clusters...` let me just assure you that it's fine. :sweat_smile: It takes more than a matching audience for a token to be valid (the signature being the primary layer of protection here).